.travis.yml

```diff
@@ -6,11 +6,17 @@ julia:
   - nightly
 notifications:
   email: false
+matrix:
+  allow_failures:
+  - julia: nightly
 script:
   - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
   - julia -e 'Pkg.clone(pwd()); Pkg.build("Gadfly");'
   - julia -e 'Pkg.checkout("Compose")'
-  - julia --check-bounds=yes -e 'Pkg.test("Gadfly", coverage=true)'
+  # because of a bug in Julia 0.6.2 we temporarily use --compilecache=no.
+  # The bug is fixed on master and will be included in 0.6.3 so when it
+  # has been releases, we can bump the requirement and remove the flag.
+  - julia --check-bounds=yes --compilecache=no -e 'Pkg.test("Gadfly", coverage=true)'
```

Comment:
it was [suggested](https://github.com/GiovineItalia/Gadfly.jl/pull/1090#issuecomment-372654098) that some temporary methods could be added which would alleviate the need to use `compilecache`.  do you think that's possible?  if so, that would be a much better solution.

Reply:
This happens to methods in `IndirectArrays` as well as `WeakRefStrings` so we'd have to add temporary methods to at least two packages even though the methods shouldn't be needed and the bug can only be triggered by Gadfly. Releasing a version that will cause ambiguity errors is also really bad though. We only have bad options.

Reply:
Let me try to add the methods to the two packages. If tests pass with the methods defined, it might be worth doing even though I'd much prefer a 0.6.3 release to doing this.

Reply:
Meanwhile, I think it would make sense to merge this as is but holding off on a new release until we have looked more into this issue.

Reply:
What are the problematic methods? Possibly related: https://github.com/JuliaData/WeakRefStrings.jl/pull/26

Reply:
Yes. That might be one of them. I'll try to check out master of WeakRefStrings and see how it goes.

Reply:
With WeakRefString master and https://github.com/JuliaArrays/IndirectArrays.jl/pull/13 all Gadfly tests pass with precompilation enabled. So it looks like we can actually work around this in a reasonable way.